packages/ethereum-contracts/test/foundry/FoundrySuperfluidTester.sol

```diff
@@ -82,136 +167,1290 @@ contract FoundrySuperfluidTester is Test {
             superToken.upgrade(INIT_SUPER_TOKEN_BALANCE);
             _expectedTotalSupply += INIT_SUPER_TOKEN_BALANCE;
             vm.stopPrank();
+            _helperTakeBalanceSnapshot(superToken, TEST_ACCOUNTS[i]);
+        }
+    }
+
+    /// @notice Deploys a Native Asset SuperToken and gives tokens to the test accounts
+    /// @dev We use vm.deal to give each account a starting amount of ether
+    function _setUpNativeAssetSuperToken() internal {
+        (superToken) = sfDeployer.deployNativeAssetSuperToken("Super ETH", "ETHx");
+        for (uint256 i = 0; i < N_TESTERS; ++i) {
+            vm.startPrank(TEST_ACCOUNTS[i]);
+            vm.deal(TEST_ACCOUNTS[i], INIT_TOKEN_BALANCE);
+            ISETH(address(superToken)).upgradeByETH{ value: INIT_SUPER_TOKEN_BALANCE }();
+            _expectedTotalSupply += INIT_SUPER_TOKEN_BALANCE;
+            vm.stopPrank();
+            _helperTakeBalanceSnapshot(superToken, TEST_ACCOUNTS[i]);
+        }
+    }
+
+    /// @notice Deploys a Pure SuperToken and gives tokens to the test accounts
+    /// @dev This contract receives the total supply then doles it out to the test accounts
+    function _setUpPureSuperToken() internal {
+        uint256 initialSupply = INIT_SUPER_TOKEN_BALANCE * N_TESTERS;
+        (superToken) = sfDeployer.deployPureSuperToken("Super MR", "MRx", initialSupply);
+        _expectedTotalSupply = initialSupply;
+        for (uint256 i = 0; i < N_TESTERS; ++i) {
+            superToken.transfer(TEST_ACCOUNTS[i], INIT_SUPER_TOKEN_BALANCE);
+            _helperTakeBalanceSnapshot(superToken, TEST_ACCOUNTS[i]);
+        }
+    }
+
+    /// @notice Deploys a SuperToken based on the testSuperTokenType selected
+    function _setUpSuperToken() internal {
```

Comment:
okay with this ergonomics? how is the use sites look like?

Reply:
using enums instead